File: content/how-to/account/examples/streaming.js
const Alpaca = require('@alpacahq/alpaca-trade-api')
const alpaca = new Alpaca()

// Prepare the websocket connection and subscribe to account_updates.
alpaca.websocket.onConnect(function () {
    console.log("Connected")
    client.subscribe(['account_updates'])
    setTimeout(() => {
        client.disconnect()
    }, 30 * 1000)
})
alpaca.websocket.onDisconnect(() => {
    console.log("Disconnected")
})
alpaca.websocket.onStateChange(newState => {
    console.log(`State changed to ${newState}`)
})

// Handle incoming account updates.
alpaca.websocket.onAccountUpdate(accountDdata => {
    // Track the cash balance in our account.
    const balance = accountDdata['cash']
    console.log(`Update for account. Cash balance: ${balance}`)
})

// Start listening for updates.
alpaca.websocket.connect()

File: content/how-to/account/examples/streaming.py
import alpaca_trade_api as tradeapi

conn = tradeapi.stream2.StreamConn()

# Handle incoming account updates.
# The r indicates that we're listening for a regex pattern.
@conn.on(r'.*')
async def on_msg(conn, channel, data):
    # Track the cash balance in our account.
    print("Update for account. Cash balance: {}".format(data['cash']))

# Start listening for updates.
conn.run(['account_updates'])